packages/app/src/systems/Error/hooks/useReportError.tsx

```diff
@@ -0,0 +1,98 @@
+import { ReportErrorFrequency } from '@fuel-wallet/types';
+import { useSelector } from '@xstate/react';
+
+import { type ReportErrorMachineState } from '../machines';
+
+import { REPORT_ERROR_FREQUENCY_KEY } from '~/config';
+import { Services, store } from '~/store';
+import { useStorageItem } from '~/systems/Core/hooks/useStorage';
+
+const selectors = {
+  hasErrorsToReport(state: ReportErrorMachineState) {
+    return state.context.hasErrors;
+  },
+  isLoadingSendOnce(state: ReportErrorMachineState) {
+    return (
+      state.hasTag('loading') &&
+      state.context.frequency === ReportErrorFrequency.ONCE
+    );
+  },
+  isLoadingSendAlways(state: ReportErrorMachineState) {
+    return (
+      state.hasTag('loading') &&
+      state.context.frequency === ReportErrorFrequency.ALWAYS
+    );
+  },
+  isLoadingDontSend(state: ReportErrorMachineState) {
+    return (
+      state.hasTag('loading') &&
+      state.context.frequency === ReportErrorFrequency.DONT
+    );
+  },
+};
+
+export function useReportError() {
+  const [reportErrorFrequency, setReportErrorFrequency] =
+    useStorageItem<ReportErrorFrequency>(
+      REPORT_ERROR_FREQUENCY_KEY,
+      ReportErrorFrequency.ONCE
+    );
+  const service = store.useService(Services.reportError);
+  store.useUpdateMachineConfig(Services.reportError, {
+    context: {
+      frequency: reportErrorFrequency || ReportErrorFrequency.ONCE,
+    },
+    actions: {
+      reload: () => {
+        store.closeOverlay();
+      },
```

Comment:
- reload is not actually reloading, should change to closeDialog ?
- I see there is an option to report error silently. on this case other dialog being used can be closed incorrectly?